tests/core/test_agent.py

```diff
@@ -163,6 +163,18 @@ def test_agent_wrong_use_of_load(tmpdir, default_domain):
     assert agent.model_directory is not None
 
 
+async def test_agent_update_model_none_domain(trained_model):
+    agent = await load_agent(model_path=trained_model)
+    agent.update_model(
+        None,
+        agent.policy_ensemble,
+        agent.fingerprint,
+        agent.interpreter,
+        agent.model_directory,
+    )
+    assert agent.domain is None
```

Comment:
ideally we should also test that such an agent is functional - i.e. loads an NLU-only model and is able to parse an intent message (with `domain` and `policy_ensemble` set to `None`)

Reply:
In `update_model`'s signature, the `policy_ensemble` is not marked as optional. Should I make it optional so that I can pass `None`? Setting it to that would make `is_ready()` always return `False`, is that something I should change as well?

Reply:
@ricwo should I use `handle_text` or `handle_message` on the agent? Should these methods return `None` or something else?

I've tried using `handle_message` but the result is always `None`. I believe that is because `TrackerStore.init_tracker()` always returns `None` if no domain is set. If I change it to always return a `DialogueStateTracker`, then another issue appears on `MemoizationPolicy.predict_action_probabilities()`, on line 198, where it is assumed that `domain` is present. The method raises an exception on when it tried to do `[0.0] * domain.num_actions`.

Reply:
I guess for full NLU-only support we need to make another change: the tracker stores need to be able to function without a domain, and this would also imply that an agent can have `is_ready()=True` if no policy ensemble is loaded.

> should I use handle_text or handle_message on the agent? Should these methods return None or something else?
you can use either. the agent should recognise when an NLU-only model is loaded and print a warning

> I believe that is because TrackerStore.init_tracker() always returns None if no domain is set
we should in principle allow tracker fetching and creation without the domain (we can just pass `None` for the slots argument if there's no domain)

Reply:
@ricwo Ok, will try to change that, then. How would `Agent.predict_next_action()` need to be modified to work without a domain and policy ensemble?

Reply:
I ended up making `Agent.handle_message()` exit right before processing actions, if the domain or the policy ensemble are not set. A warning is also printed. The test now checks that the `UserUttered` event contains the correct intent information after having received a message.